src/katgpucbf/xbgpu/main.py

```diff
@@ -48,13 +49,83 @@
 from ..spead import DEFAULT_PORT
 from ..utils import add_signal_handlers, parse_source
 from .correlation import device_filter
+from .output import BengineOutput, XengineOutput
 
 logger = logging.getLogger(__name__)
 
 
+class BengineOutputDict(TypedDict, total=False):
+    """Configuration options for a beamformer output.
+
+    Unlike :class:`BengineOutput`, all the fields are optional, so that it can
+    be built up incrementally. They must all be filled in before using it to
+    construct a :class:`BengineOutput`.
+    """
+
+    beams: int
+    dst: list[Endpoint]
+    channels_per_substream: int
+    spectra_per_heap: int
+    send_rate_factor: float
+
+
+def parse_bengine(value: str) -> BengineOutputDict:
+    """Parse a string with B-engine configuration data.
+
+    The string has a comma-separated list of key=value pairs. See
+    :class:`BengineOutput` for the valid keys and types. The following
+    keys are required:
+
+    - beams
+    - channels_per_substream
```

Comment:
I'm not seeing how channels_per_substream can be configured for an output. We receive some number of channel as input, and the same set of channels are in the output.

Reply:
Again, I think this points to my internally-broken telephone where, as per [this comment on the Jira ticket](https://skaafrica.atlassian.net/browse/NGC-905?focusedCommentId=191493)
* I ended up looking at the literal output data dimensions/attributes needed to describe the output.
* In hindsight, that does seem strange, those attributes would already be part of the Engine (or even eventual Pipeline [by proxy]).